src/types.rs

```diff
@@ -186,6 +186,63 @@ pub struct SdoIndex {
     pub subindex: u8,
 }
 
+#[derive(Debug, Clone)]
+pub struct SdoInfo {
+    pub slave_position: u16,
+    pub position: u16,
+    pub index: u16,
+    pub max_subindex: u8,
+    pub object_code: u8,
+    pub name: String,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum Access {
+    Unknown,
+    ReadOnly,
+    WriteOnly,
+    ReadWrite,
+}
+
+#[derive(Debug, Clone)]
+pub struct SdoEntryAccess {
+    pub pre_op: Access,
+    pub safe_op: Access,
+    pub op: Access,
+}
+
+impl From<(u8, u8)> for Access {
+    fn from((read, write): (u8, u8)) -> Self {
+        match (read, write) {
+            (0, 1) => Access::ReadOnly,
+            (1, 0) => Access::WriteOnly,
+            (1, 1) => Access::ReadWrite,
+            _ => Access::Unknown,
```

Comment:
@birkenfeld Do you know what it means if e.g. both are zero? Is it unknown or not accessible at all?

Reply:
I don't know, but looking at the master code:

* the access mode is read directly from the device
* the mode doesn't seem to be checked by the master

So a badly implemented device could presumably return (0,0) for the access but still process read and/or write requests when the respective ioctls are called.

Reply:
ok, so would you agree with the `Unknown` variant here?

Reply:
Yes!